src/main/java/org/datadog/jenkins/plugins/datadog/transport/HttpClient.java

```diff
@@ -0,0 +1,10 @@
+package org.datadog.jenkins.plugins.datadog.transport;
+
+public interface HttpClient {
+
+    void send(PayloadMessage obj);
+
+    void stop();
+
+    void close();
+}
```

Comment:
Is this plugin JRE8+ only? If so, you might want to consider using the async stuff that was added in Java8 (CompletableFuture & co). That would allow you to:

- use your executor's threads more efficiently through the use of non-blocking http calls
- not have to manage a manual queue of work (it would be managed by the executor)
- allow the callers to decide what to do if their request failed

Reply:
As we discussed offline, the best approach here had been to use the `java.net.http.HttpClient` to implement the HTTP calls asynchronously, but unfortunately, this class is only available since JDK11. 

So this change would need an external library, which is something that could introduce issues in terms of dependency version clashes. To reduce the risk, we will use the approach based on `java.net.url.HttpURLConnection`.